test/Microsoft.Extensions.Internal.Test/ObjectMethodExecutorTest.cs

```diff
@@ -427,6 +511,17 @@ public ValueTask<string> ValueTaskOfReferenceType(string result)
             public void MethodWithMultipleParameters(int valueTypeParam, string referenceTypeParam)
             {
             }
+
+            public FSharpAsync<string> FSharpAsyncMethod(string parameter)
```

Comment:
I will add an F# test project to the MVC repo and put in a functional test there. That will be a separate PR (since it's a different repo).

Initially it will only cover this specific functionality (invoking actions that return `FSharpAsync<T>`) but in the future we can add whatever extra functionality and tests we want.